sw/host/opentitanlib/src/transport/ti50emulator/uart.rs

```diff
@@ -56,15 +50,14 @@ impl Ti50Uart {
     }
 
     pub fn get_socket(&self) -> Result<RefMut<UnixStream>> {
-        self.reconnect()?;
```

Comment:
I see that you are moving this to spawn_process(...)
What command OTT was used to tested this changes.

Reply:
I tested this by this sequence of commands (from within ti50/common), first:
```
$ cp build/system_test_auto/host_emulation/host_emulation/image.A.bin /tmp/system_test_auto.bin && /mnt/host/source/src/third_party/lowrisc/opentitan/bazel-bin/sw/host/opentitansession/opentitansession --foreground --rcfile= --logging=warn --conf ports/host_emulation/software/tools/host_emulation.json --conf ports/host_emulation/software/tools/ti50_host_emulation.json --interface ti50emulator --executable-directory /tmp/ --executable system_test_auto.bin
```
Then in a separate window, I did:
```
$ /mnt/host/source/src/third_party/lowrisc/opentitan/bazel-bin/sw/host/opentitantool/opentitantool --rcfile= --format=json --interface=proxy --proxy=127.0.0.1 --port 9900 console -q --uart console
```
Finally, I released the emulator by:
```
$ /mnt/host/source/src/third_party/lowrisc/opentitan/bazel-bin/sw/host/opentitantool/opentitantool --rcfile= --format=json --interface=proxy --proxy=127.0.0.1 --port 9900 gpio write RESET true
```

What I noted was that with the PR, the console in the second command above would show me all output, starting from `Starting RO 1.2.3 RW 1.2.3`.  Without this PR, a lot of the initial output is omitted, causing tests to not be able to reliably recognize e.g. `##KERNEL TESTS START` in the output.

I also tried setting the `RESET` to `false` and `true` again multiple times, each time seeing output from the new sub-process.

Reply:
Thanks i will check the result on my side and let you know the outcome.

Reply:
LGTM